libavfilter/vf_drawtext.c

```diff
@@ -1536,13 +1583,27 @@ static int filter_frame(AVFilterLink *inlink, AVFrame *frame)
     s->var_values[VAR_PKT_SIZE] = frame->pkt_size;
     s->metadata = frame->metadata;
 
-    draw_text(ctx, frame, frame->width, frame->height);
+    for (int i = 0; i < loop; i++) {
+        if (header) {
+            bbox = av_get_detection_bbox(header, i);
+            strcpy(s->text, bbox->detect_label);
+            for (int j = 0; j < bbox->classify_count; j++) {
+                strcat(s->text, ", ");
+                strcat(s->text, bbox->classify_labels[j]);
+            }
+            s->x = bbox->x;
+            s->y = bbox->y - s->fontsize;
+        }
+        draw_text(ctx, frame, frame->width, frame->height);
+    }
 
     av_log(ctx, AV_LOG_DEBUG, "n:%d t:%f text_w:%d text_h:%d x:%d y:%d\n",
            (int)s->var_values[VAR_N], s->var_values[VAR_T],
            (int)s->var_values[VAR_TEXT_W], (int)s->var_values[VAR_TEXT_H],
            s->x, s->y);
 
+
+    av_freep(&s->text);
```

Comment:
will there be double-free for the original function?

Reply:
i did not notice any free of this mem, it is still OK to pass NULL pointer to av_freep().

Reply:
In the original function, s->text is allocated/freed at other places, I'm not sure if this free conflicts.